### PR TITLE
RFC: Export iszero

### PR DESCRIPTION
**Pull request by ararslan**
As mentioned by @stevengj in #19947, introduced in #17623.

### CODE REVIEW

**Comment by tkelman** [comment]
if it's exported, it needs to be documented and the docs should be in the stdlib, if they aren't yet

**Comment by ararslan** [comment]
The web UI is cumbersome for this stuff, particularly for editing multiple files, so I'll improve this a little later. I can also add a testset for it.

**Comment by ararslan** [comment]
Now with 200% more docs and tests!

**Comment by ararslan** [comment]
Does this need a NEWS item?

**Comment by stevengj** [comment]
Yes, a NEWS item would be good.

**Comment by stevengj** [comment]
(I was thinking that `iszero(A::Array{T})` could probably be made faster for `T` one of the basic numeric types, since you just need to [search for a nonzero byte](http://stackoverflow.com/questions/2589736/fast-way-to-check-if-an-array-of-chars-is-zero) and could probably arrange it to use SIMD instructions on 256-bit chunks, but it's maybe not worth the trouble.  )

**Comment by ararslan** [comment]
That would be pretty neat. I could take a crack at that at some point, but it would probably be better as a separate PR so that this and #19949 can be merged in time for 0.6.

**Review by Sacha0 — APPROVED** [review]

File: NEWS.md [hunk 1]
```diff
@@ -159,6 +159,8 @@ Library improvements
   * `logging` can be used to redirect `info`, `warn`, and `error` messages
     either universally or on a per-module/function basis ([#16213]).
 
+  * `iszero` is now exported from `Base` ([#19950]).
```
Comment (stevengj) [inline NEWS.md]: This assumes that people already know what this function is.
```
New `iszero(x)` function to quickly check whether `x` is zero (or is all zeros, for an array) ([#19950]).
```

**Comment by ararslan** [comment]
Actually, perhaps that optimization could be made in `any`/`all`, since the array-reducing method for `iszero` is just `all(iszero, x)`.

File: NEWS.md (hunk 1 above)
Comment (ararslan) [inline NEWS.md]: Much, much better. I've gone with that. Thanks!

**Review by stevengj — APPROVED** [review]

File: test/numbers.jl [hunk 2]
```diff
@@ -2864,3 +2864,19 @@ let types = (Base.BitInteger_types..., BigInt, Bool)
 end
 
 @test !isempty(complex(1,2))
+
+@testset "iszero" begin
+    # Numeric scalars
+    for T in Iterators.flatten(subtypes.([AbstractFloat, Signed, Unsigned]))
```
Comment (tkelman) [inline test/numbers.jl]: these types should probably just be written out - there's a type `SOE.Sgnd` in the core test that causes a failure here if it runs in the same worker

**Pull request merged**